numpy/lib/arraypad.py

```diff
@@ -797,9 +799,22 @@ def pad(array, pad_width, mode='constant', **kwargs):
         # Create a new zero padded array
         padded, _ = _pad_simple(array, pad_width, fill_value=0)
         # And apply along each axis
-        for axis in range(padded.ndim):
-            np.apply_along_axis(
-                function, axis, padded, pad_width[axis], axis, kwargs)
+
+        in_dims = list(range(array.ndim))
+        for axis in in_dims:
+            # Iterate using ndindex as in apply_along_axis, but assuming that
+            # function operates inplace on the padded array.
+
+            # view with the iteration axis at the end
+            view = transpose(padded,
```

Comment:
Perhaps worth a comment with the equivalent `moveaxis` code (and a note that it would be slower)

Reply:
`moveaxis` calls `transpose` internally in a similar fashion to what is proposed here.  `moveaxis` also returns a view, so performance should be very similar. I couldn't measure a consistent difference using `%timeit` on the docstring example and the difference should be even more negligible at larger sizes. This makes sense because `moveaxis` is only called once per axis while `function` is typically called many times per axis.

Given that
```Python
view = moveaxis(padded, axis, -1)
```

is easier to read than
```Python
view = transpose(padded,
                 in_dims[:axis] + in_dims[axis + 1:] + [axis])
```
I just switched to using `moveaxis` in the latest commit. Perhaps `transpose` was used internally for `apply_along_axis` mainly because that function predates the introduction of `moveaxis`?

Reply:
Thanks for measuring. My thinking was that since you already had the tuple of all axes constructed, and knew they were all in range, you could avoid `moveaxis` reconstructing the tuple and doing range checks each time. Sounds like the gain is so minimal that clarity is more important.

Reply:
Yes, I think we are on the same page. I initially implemented your suggested comment, but eventually decided it was clearer to just call `moveaxis`.